pkg/plugins/mellanox/mellanox_plugin.go

```diff
@@ -177,70 +175,92 @@ func runCommand(command string, args ...string) (string, error) {
 	return stdout.String(), err
 }
 
-func getMlnxNicFwData(pciAddress string) (*mlnxNic, error) {
+func getMlnxNicFwData(pciAddress string) (current, next *mlnxNic, err error) {
 	glog.Infof("mellanox-plugin getMlnxNicFwData(): for device %s", pciAddress)
-	fwData := &mlnxNic{}
+	err = nil
 	attrs := []string{TotalVfs, EnableSriov}
 	singlePort, err := isSinglePortNic(pciAddress)
 	if err != nil {
-		return nil, err
+		glog.Errorf("mellanox-plugin getMlnxNicFwData(): %v", err)
+		return
 	}
+	firstPort := isFirstPort(pciAddress)
+
 	if singlePort {
 		attrs = append(attrs, "LINK_TYPE")
-		fwData.singlePort = true
 	} else {
-		fwData.singlePort = false
-		if isFirstPort(pciAddress) {
+		if firstPort {
 			attrs = append(attrs, "LINK_TYPE_P1")
```

Comment:
minor:
you can use:
linkTypeName = fmt.Sprintf("LINK_TYPE_P%d", pf_function + 1 )